core/src/main/java/ai/timefold/solver/core/config/solver/monitoring/SolverMetric.java

```diff
@@ -90,12 +92,15 @@ public enum SolverMetric {
         this.isConstraintMatchBased = isConstraintMatchBased;
     }
 
-    public String getMeterId() {
+    public @NonNull String getMeterId() {
         return meterId;
     }
 
-    public static void registerScoreMetrics(SolverMetric metric, Tags tags, ScoreDefinition<?> scoreDefinition,
-            Map<Tags, List<AtomicReference<Number>>> tagToScoreLevels, Score<?> score) {
+    public static void registerScoreMetrics(@NonNull SolverMetric metric, @NonNull Tags tags,
+            // TODO: @NonNull Map<@NonNull Tags, @NonNull List<@NonNull AtomicReference<@NonNull Number>>> tagToScoreLevels?
+            @NonNull ScoreDefinition<?> scoreDefinition,
+            @NonNull Map<@NonNull Tags, List<AtomicReference<Number>>> tagToScoreLevels,
+            @NonNull Score<?> score) {
```

Comment:
@triceo does it make sense to have `@NonNull Map<@NonNull Tags, @NonNull List<@NonNull AtomicReference<@NonNull Number>>>` as type for `tagToScoreLevels`?

Reply:
Not our best API design here. :-(

You know Kotlin better than I do. Would these annotations on this crazy type actually help? I feel like they would, but I will defer to your judgment here.

Reply:
I think it makes sense to use the annotation too - when unwrapping the mapped value, things might get messy: 

```
scoreLevels?.forEach { it?.get()?.times(1) }
```

Especially when one for whatever reason would have to handle the null-case here. That being said, the safe-calls wouldn't be mandatory here, since all the types here would be platform types. Still, it's better to be certain as a user, and with the correct annotations its explicitly stated in the code.

Reply:
And I fear I've seen far worse API designs 😅